test/utils.dart

```diff
@@ -0,0 +1,95 @@
+library heron.test.utils;
+
+import 'dart:async';
+import 'dart:io';
+import 'package:scheduled_test/descriptor.dart';
+import 'package:scheduled_test/scheduled_test.dart';
+import 'package:scheduled_test/scheduled_process.dart';
+import 'package:scheduled_test/descriptor.dart' as d;
+import 'package:path/path.dart' as p;
+
+import 'dart:mirrors';
+
+export 'sample_descriptors.dart';
+
+/// Produces the path to heron executable script
+String get _heronPath =>
+    p.normalize(p.join(p.dirname(currentMirrorSystem()
+        .findLibrary(#heron.test.utils)
+        .uri
+        .toFilePath()), '../'));
+
+/// Schedules heron execution in sandbox
+ScheduledProcess runHeron({Future inDirectory}) {
+  if (inDirectory == null) {
+    throw new ArgumentError.notNull('inDirectory');
+  }
+  String heronExec = p.join(_heronPath, 'bin', 'heron.dart');
+  return new ScheduledProcess.start(
+      Platform.resolvedExecutable, ['--checked', heronExec], workingDirectory: inDirectory,
+      description: 'Executing heron');
+}
+
+_DeferredValue<String> _sandbox;
+
+Future<String> get sandboxPath => _sandbox.future;
+
+/// Creates a temporary directory the tests can spam into
+///
+/// if [cleanUpAfterTests] is `false`, the temporary directory created for sandbox
+/// will not be deleted.
+void setupSandbox({bool cleanUpAfterTests: true}) {
```

Comment:
This could be spared with a wrapper shell script that removes the temp directory after the test was running. I don't mind adding it here, but I'd like to ask you to add a note in the comment.